dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/grpc/GrpcCompositeCodecFactory.java

```diff
@@ -22,19 +22,14 @@
 import org.apache.dubbo.remoting.http12.message.HttpMessageDecoderFactory;
 import org.apache.dubbo.remoting.http12.message.HttpMessageEncoderFactory;
 import org.apache.dubbo.remoting.http12.message.MediaType;
-import org.apache.dubbo.remoting.utils.UrlUtils;
 import org.apache.dubbo.rpc.model.FrameworkModel;
 
 @Activate
 public class GrpcCompositeCodecFactory implements HttpMessageEncoderFactory, HttpMessageDecoderFactory {
 
     @Override
     public HttpMessageCodec createCodec(URL url, FrameworkModel frameworkModel, String mediaType) {
-        String serializeName = UrlUtils.serializationOrDefault(url);
-        WrapperHttpMessageCodec wrapperHttpMessageCodec = new WrapperHttpMessageCodec(url, frameworkModel);
-        wrapperHttpMessageCodec.setSerializeType(serializeName);
-        ProtobufHttpMessageCodec protobufHttpMessageCodec = new ProtobufHttpMessageCodec();
-        return new GrpcCompositeCodec(protobufHttpMessageCodec, wrapperHttpMessageCodec);
+        return new GrpcCompositeCodec(url, frameworkModel, mediaType);
```

Comment:
**Optimization Issue**: The creation of GrpcCompositeCodec now directly uses URL, FrameworkModel, and mediaType, potentially bypassing detailed configuration or customization that was previously possible with individual codec setup. <br> **Fix**: Ensure that the GrpcCompositeCodec constructor properly handles all necessary configurations and customizations that might have been done in the individual codecs. Consider adding methods to GrpcCompositeCodec to allow for detailed configuration. <br> **Code Suggestion**: 
 ```
 -        return new GrpcCompositeCodec(url, frameworkModel, mediaType);
 +        GrpcCompositeCodec codec = new GrpcCompositeCodec(url, frameworkModel, mediaType);
 +        // Add necessary configuration methods on codec if needed
 +        return codec;
 ```